server/tests/src/test/java/org/infinispan/server/persistence/TableManipulation.java

```diff
@@ -0,0 +1,115 @@
+package org.infinispan.server.persistence;
+
+import java.sql.Connection;
+import java.sql.Driver;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Enumeration;
+
+import org.infinispan.commons.marshall.ProtoStreamMarshaller;
+import org.infinispan.commons.marshall.WrappedByteArray;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.persistence.jdbc.JdbcUtil;
+import org.infinispan.persistence.jdbc.configuration.PooledConnectionFactoryConfiguration;
+import org.infinispan.persistence.jdbc.configuration.PooledConnectionFactoryConfigurationBuilder;
+import org.infinispan.persistence.jdbc.connectionfactory.ConnectionFactory;
+import org.infinispan.persistence.jdbc.impl.connectionfactory.PooledConnectionFactory;
+import org.infinispan.persistence.keymappers.DefaultTwoWayKey2StringMapper;
+
+import com.mysql.cj.jdbc.AbandonedConnectionCleanupThread;
+
+public class TableManipulation {
```

Comment:
If we adapt this to `implements AutoCloseable`, we can move the `releaseConnection` logic to the `stop()` implementation and then just do the following when using the code:

```java
try (TableManipulation tm = new TableManipulation(..)) {
 // Test logic
}
```